src/main/scala/com/mozilla/telemetry/streaming/ErrorAggregator.scala

```diff
@@ -375,6 +376,7 @@ object ErrorAggregator {
     val outputPath = opts.outputPath()
 
     aggregate(pingsDataframe, raiseOnError = opts.raiseOnError(), online = false)
+      .repartition(numPartitions)
```

Comment:
So here we say we want 600 partitions and below we say we want to partition by day. That means we will likely have empty partitions unless we run the batch job on 600 days, right?

Reply:
Nope, this gives 600 partitions per day.

Reply:
Is it 600 partitions in total or 600 per day?

Reply:
Proof :)

```
Welcome to Scala 2.11.8 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_102).
Type in expressions for evaluation. Or try :help.

scala> import org.apache.spark.sql.SparkSession
import org.apache.spark.sql.SparkSession

scala> val spark = SparkSession.builder().master("local[1]").getOrCreate()
17/09/15 08:51:20 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
spark: org.apache.spark.sql.SparkSession = org.apache.spark.sql.SparkSession@349a6105

scala> import spark.implicits._
import spark.implicits._

scala> List(("a", 1), ("a", 2), ("b", 1), ("b", 2)).toDF("id", "val")
res0: org.apache.spark.sql.DataFrame = [id: string, val: int]

scala> List(("a", 1), ("a", 2), ("b", 1), ("b", 2)).toDF("id", "val").repartition(2).write.partitionBy("id").parquet("/tmp/test-output")
[Stage 1:>                                                          (0 + 1) / 2]SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.

scala> import scala.sys.process._
import scala.sys.process._

scala> "ls -R /tmp/test-output"!
warning: there was one feature warning; re-run with -feature for details
_SUCCESS
id=a
id=b

/tmp/test-output/id=a:
part-00000-59e8f480-6975-4527-89f1-3316cf42a186.snappy.parquet
part-00001-59e8f480-6975-4527-89f1-3316cf42a186.snappy.parquet

/tmp/test-output/id=b:
part-00000-59e8f480-6975-4527-89f1-3316cf42a186.snappy.parquet
part-00001-59e8f480-6975-4527-89f1-3316cf42a186.snappy.parquet
res8: Int = 0
```

Reply:
This is great, thanks!